rllib/offline/dataset_reader.py

```diff
@@ -228,14 +228,19 @@ def __init__(self, ds: ray.data.Dataset, ioctx: Optional[IOContext] = None):
                     if not self._ioctx.config.get("_disable_preprocessors", False)
                     else None
                 )
-            self._dataset.random_shuffle(seed=seed)
             print(
                 f"DatasetReader {self._ioctx.worker_index} has {ds.count()}, samples."
             )
             # TODO: @avnishn make this call seeded.
             # calling random_shuffle_each_window shuffles the dataset after
             # each time the whole dataset has been read.
-            self._iter = self._dataset.repeat().random_shuffle_each_window().iter_rows()
+
+            def iterator():
+                while True:
+                    ds = self._dataset.random_shuffle()
+                    yield from ds.iter_rows()
```

Comment:
An alternative to calling `random_shuffle()` each epoch is to use `local_shuffle_buffer`, but because we don't know what the data looks like, it's hard to configure it correctly.

Reply:
Is `local_shuffle_buffer ` expected yield higher throughput?

Reply:
Yeah, but I don't know how you'd configure it without knowing what the data looks like. If you set it too low, you might get too little randomness.

Reply:
Let's just use `random_shuffle` here to maintain the same behavior as before. If needed, we can change to local shuffle later in a separate PR. Let's focus on code deprecation here. Thanks.